pkg/actions/mustgather.go

```diff
@@ -0,0 +1,364 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package actions
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/eclipse/codewind-installer/pkg/appconstants"
+	"github.com/eclipse/codewind-installer/pkg/docker"
+	"github.com/eclipse/codewind-installer/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+var codewindHome = filepath.Join(homeDir, ".codewind")
+var nowTime = time.Now().Format("20060102150405")
+var mustGatherDirName = filepath.Join(codewindHome, "mustgather", nowTime)
+
+var isLoud = true
+
+func logMG(input ...string) {
+	if isLoud {
+		fmt.Println(input[0])
+	}
+}
+
+//MustGatherCommand to gather logs and project files to aid diagnosis of Codewind errors
+func MustGatherCommand(c *cli.Context) {
```

Comment:
This function is a bit on the big side. Could it be split up into parts for each stage? For example the Zipping at the end could go into pkg/utils/filesystem.go which already has an `UnZip()` function to unzip a file to a destination.

Reply:
Sure - I was trying to keep the main function having content as opposed to just being a list of function calls, but I can see how that might make it clearer.